test/miner-test.js

```diff
@@ -180,4 +181,81 @@ describe('Miner', function() {
       assert(txids.has(block.txs[i].hash()));
     }
   });
+
+   it('should not include free transaction in a block', async () => {
+    // Miner does not have any room for free TXs
+    miner.options.minWeight = 0;
+
+    const value = 1 * 1e6;
+    const fee = 0;
+
+    // Get a change address
+    const change = wallet.createChange().getAddress();
+    const mtx = new MTX();
+    coin = wallet.getCoins()[0];
+    mtx.addCoin(coin);
+    mtx.addOutput(addr, value);
+    mtx.addOutput(change, coin.value - value - fee); // no fee
+    wallet.sign(mtx);
+    parentTX = mtx.toTX();
+
+    await mempool.addTX(parentTX, -1);
+    assert.strictEqual(mempool.map.size, 1);
+
+    const block = await miner.mineBlock(chain.tip, addr);
+    await chain.add(block);
+
+    // TX is still in mempool, nothing in block except coinbase
+    assert.strictEqual(mempool.map.size, 1);
+    assert.strictEqual(block.txs.length, 1);
+  });
+
+  it('should fail to double spend the coin - duplicate tx in mempool', async () => {
+    const value = 1 * 1e6;
+    const fee = 1000;
+    const mtx = new MTX();
+
+    const change = wallet.createChange().getAddress();
+    mtx.addCoin(coin);
+    mtx.addOutput(addr, value);
+    mtx.addOutput(change, coin.value - value - fee);
+    wallet.sign(mtx);
+    const tx = mtx.toTX();
+
+    assert.rejects(
+      async () => await mempool.addTX(tx, -1),
+      {
+        code: 'duplicate',
+        reason: 'bad-txns-inputs-spent'
+      }
+    );
+    // orignal tx is still in mempool
+    assert.strictEqual(mempool.map.size, 1);
+  });
```

Comment:
This looks more like a mempool test than a miner test. Do we need to add it? Is this error already covered in the mempool tests?